src/jetstream/config/config.go

```diff
@@ -133,49 +129,22 @@ func SetStructFieldValue(value reflect.Value, field reflect.Value, val string) e
 	return nil
 }
 
-// IsSet - is the specified config name set?
-func IsSet(name string) bool {
-	_, err := GetValue(name)
-	return err == nil
-}
-
-// GetString -  Get the string value for the named configuration property
-func GetString(name string) string {
-	v, _ := GetValue(name)
-	return v
-}
-
-// GetValue tries to look up an env var of the given name and then
-// tries to look up the secret file of the same
-func GetValue(name string) (string, error) {
-	env := os.Getenv(name)
-	if len(env) == 0 {
-		env = loadedConfig[name]
-	}
-
-	if len(env) == 0 {
-		return readSecretFileTestHarness(name)
-	}
-
-	return env, nil
-}
-
-var readSecretFileTestHarness = readSecretFile
-
-// readSecretFile reads a variable in the form HELLO_THERE from a file
+// NewSecretsDirLookup - create a secret dir lookup
+// reads a variable in the form HELLO_THERE from a file
 // in /etc/secrets/hello-there
-func readSecretFile(name string) (string, error) {
-	name = strings.ToLower(strings.Replace(name, "_", "-", -1))
-	filename := filepath.Join(secretsDir, name)
-
-	contents, err := ioutil.ReadFile(filename)
-	if os.IsNotExist(err) {
-		return "", notFoundErr(filename)
-	} else if err != nil {
-		return "", fmt.Errorf("failed to read secret file %q: %v", name, err)
-	}
+func NewSecretsDirLookup(secretsDir string) env.Lookup {
+	return func(name string) (string, bool) {
+		name = strings.ToLower(strings.Replace(name, "_", "-", -1))
+		filename := filepath.Join(secretsDir, name)
+
+		contents, err := ioutil.ReadFile(filename)
+		if err != nil {
+			log.Warnf("Error reading secrets file: %s, %s", filename, err)
```

Comment:
Secrets is the last config lookup in the chain, so any env that is looked up that does not exist prints this message - should not print error if the secrets file does not exist - just means there is no secret for the env requested.

Reply:
It should have only printed once during app startup. Either way I've suppressed this message from printing.